Source/ReferenceConnectorWorkerService/ConnectorTask.cs

```diff
@@ -309,6 +320,40 @@ private async Task SubmitBinary(string itemExternalId, CancellationToken cancell
             while (binarySubmitContext.SubmitResult.SubmitStatus == SubmitResult.Status.Deferred && tryCount < 30);
         }
 
+        private void HandleSubmitPipelineResult(SubmitContext submitContext)
+        {
+            // Check the result of the submission.
+            if (submitContext.SubmitResult.SubmitStatus == SubmitResult.Status.OK)
+            {
+                // The item was submitted successfully! Records365 guarantees that the item is now stored durably and
+                // the connector has no need to store or queue this item any more.
+            }
+            else if (submitContext.SubmitResult.SubmitStatus == SubmitResult.Status.Skipped)
+            {
+                // The item was skipped by the submit pipeline - it may have been filtered out. The connector has no need to store
+                // or queue this item any more.
+            }
+            else if (submitContext.SubmitResult.SubmitStatus == SubmitResult.Status.Deferred)
+            {
+                // The submit pipeline attempted to submit the item and the pipeline returned a result indicating that the item
+                // should be tried again later.
+            }
+            else if (submitContext.SubmitResult.SubmitStatus == SubmitResult.Status.ConnectorDisabled)
+            {
+                // The submit pipeline attempted to submit the item and the Records365 Connector API returned a response
+                // that indicated that the connector was disabled in Records365. All processing for this connector instance
+                // should stop if this result is received.
+                // Note that a user may re-enable the connector in Records365, so any book-keeping related to the connector
+                // instance should be kept. Items should be kept (dead-lettered) as well.

```

Comment:
I would separate the stashing that should be done if the Connector is disabled from the concept of dead-lettering.